src/mlpack/core/data/string_encoding_policies/bag_of_words_encoding_policy.hpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file bag_of_words_encoding_policy.hpp
+ * @author Jeffin Sam
+ * @author Mikhail Lozhnikov
+ *
+ * Definition of the BagOfWordsEncodingPolicy class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STR_ENCODING_POLICIES_BAG_OF_WORDS_ENCODING_POLICY_HPP
+#define MLPACK_CORE_DATA_STR_ENCODING_POLICIES_BAG_OF_WORDS_ENCODING_POLICY_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/data/string_encoding_policies/policy_traits.hpp>
+#include <mlpack/core/data/string_encoding.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Definition of the BagOfWordsEncodingPolicy class.
+ *
+ * BagOfWords is used as a helper class for StringEncoding. The encoder maps
+ * each dataset item to a vector of size N, where N is equal to the total number
+ * of tokens. If an item of the dataset has the i-th token, then the i-th
```

Comment:
Just to check that I understand, the total number of tokens here, if each token was a word, would be the number of unique words in the document?  If so it might be worth writing `total unique number of tokens`, but up to you.